tools/cldr-apps/js/src/views/SignCla.vue

```diff
@@ -0,0 +1,233 @@
+    />
+
+    <!-- CLA text -->
+    <a-spin v-if="loading" />
+    <div class="cla" v-html="claHtml" />
+    <hr />
+    <div>
+      Your Name:
+      <a-input :disabled="!needCla" v-model:value="userName">
+        <template #prefix>
+          <i class="glyphicon glyphicon-user" />
+        </template>
+      </a-input>
+      Your Email:
+      <a-input :disabled="!needCla" v-model:value="userEmail">
+        <template #prefix>
+          <i class="glyphicon glyphicon-envelope" />
+        </template>
+      </a-input>
+      Your Employer (or 'none'):<a-input
+        :disabled="!needCla"
+        v-model:value="userEmployer"
+      >
+        <template #prefix>
+          <i class="glyphicon glyphicon-briefcase" />
+        </template>
+      </a-input>
+      <i class="pleaseChoose" v-if="needCla && userSign == 0"
+        >Please choose one:</i
+      ><br />
+      <a-radio-group :disabled="!needCla" v-model:value="userSign">
+        <a-radio :style="radioStyle" :value="1">
+          My employer has already signed the CLA and is listed on Unicode’s
+          <a href="https://www.unicode.org/policies/corporate-cla-list/"
+            >List of Corporate CLAs</a
+          >.
+        </a-radio>
+        <a-radio :style="radioStyle" :value="2">
+          I am a self-employed or unemployed individual and I have read and
+          agree to the CLA.
+        </a-radio>
+      </a-radio-group>
+      <br />
+      <button
+        @click="sign"
+        v-if="
+          needCla &&
+          userSign != 0 &&
+          userName &&
+          userEmail &&
+          userEmployer &&
+          !readonlyCla
+        "
+      >
+        Sign
+      </button>
+      <a-alert
+        type="info"
+        v-else-if="needCla"
+        message="Please fill in the above fields."
+      />
+      <hr />
+    </div>
+    <div v-if="!needCla">
+      <a-alert
+        v-if="!readonlyCla"
+        type="warning"
+        message="If the CLA is revoked, you will not be able to continue contributing to CLDR. You will have an opportunity to re-sign if corrections are needed."
+      />
+      <button v-if="!readonlyCla" @click="revoke">Revoke</button>
+    </div>
+    <a-spin v-if="loading" />
+  </div>
+</template>
+
+<script setup>
+import * as cldrStatus from "../esm/cldrStatus.mjs";
+import * as cldrClient from "../esm/cldrClient.mjs";
+import * as cldrNotify from "../esm/cldrNotify.mjs";
+
+import { ref } from "vue";
+import { marked } from "../esm/cldrMarked.mjs";
+import claMd from "../md/cla.md";
+
+const claHtml = marked(claMd);
+
+const user = cldrStatus.getSurveyUser();
+
+if (!user) window.location.replace("#///"); // need a user
+
+let readonlyCla = ref(false);
+let needCla = ref(!user?.claSigned);
+let loading = ref(false);
+const userName = ref(user?.name);
+const userEmail = ref(user?.email);
+let userEmployer = ref(user?.org);
+let userSign = ref(0);
+const radioStyle = {
+  display: "flex",
+};
+
+async function loadData() {
```

Comment:
for separation of concerns, my preference is that Vue components shouldn't include client/server code

I realize you may not agree with that, and I don't know if I could persuade you, so I won't insist on it :-)

Reply:
This is a draft PR for the purpose of a UX / legal review. I appreciate the comments and will do them to refactor before merge. I do plan to merge with everything switched off so we can un mothball it later.

Reply:
done